### ISSUE TITLE
[BUG]: Assembly-CFG interprets globals as functions

### ISSUE THREAD

**Issue by OfekShilon** [compiler-explorer/compiler-explorer#5683]
### Describe the bug

Tested on gcc/clang, x64/arm:

The generated assembly for a non-static global like -
```
int a;
```

Looks something like
```
a:
        .long   0                               # 0x0
        .size   a, 4

        .type   b,@object                       # @b
        .globl  b
        .p2align        2, 0x0
```
(this is for clang x64, but similar for others).

This confuses the assembly-CFG into parsing `a` as a function. Likely culprit is `parser.splitToFunctions`.

### Steps to reproduce

https://godbolt.org/z/oh3scKvjE

Click the function selector dropdown in the CFG pane (see screenshot).

### Expected behavior

Function dropdown should contain just 'test():'

### Reproduction link

https://godbolt.org/z/oh3scKvjE

### Screenshots

![image](https://github.com/compiler-explorer/compiler-explorer/assets/73080/1e99aa40-f7de-41a8-9a84-d0580504dc1c)

**Comment by OfekShilon** [comment]
One suggested ~hack~ solution: drop "functions" with a body consisting entirely of assembly lines starting with a dot.  WDYT?

**Comment by jeremy-rifkin** [comment]
Thanks for opening this issue, this has been a long-standing issue but I haven't given much thought to fixing it. I the solution you're proposing sounds good to me. `getAsmDirective` might be of use for identifying the directives a little more robustly and for other assembly formats.

**Comment by OfekShilon** [comment]
Suggested fix at #5685
